server/src/main/java/org/elasticsearch/action/search/TransportSearchAction.java

```diff
@@ -1515,6 +1509,34 @@ public SearchPhase newSearchPhase(
         }
     }
 
+    /**
+     * TransportSearchAction cannot access async-search code, so can't check whether this the Task
+     * is an instance of AsyncSearchTask, so this roundabout method is used
+     * @param searchTask SearchTask to analyze
+     * @return true if this is an async search task; false if a synchronous search task
+     */
+    private boolean isAsyncSearchTask(SearchTask searchTask) {
+        assert assertAsyncSearchTaskListener(searchTask) : "AsyncSearchTask SearchProgressListener name has ";
```

Comment:
The string in the "else" clause of the search is incomplete. Probably should be "AsyncSearchTask SearchProgressListener is not one of the expected types".